Sources/PureSwiftJSON/Decoding/JSONUnkeyedDecodingContainer.swift

```diff
@@ -153,6 +140,15 @@ extension JSONUnkeyedDecodingContainer {
         )
     }
 
+    @inline(__always)
+    private func getNextValue() throws -> JSONValue {
+        guard !self.isAtEnd else {
+            throw DecodingError.dataCorruptedError(in: self,
+                                                   debugDescription: "Index \(self.currentIndex) out of bounds (\(self.array.count)) trying to decode value.")
```

Comment:
Is this how Foundation behaves? Would love to have a Foundation learning test for this behaviour, so that we ensure that we do comply.

Reply:
As it turns out, not quite, though close! Here's the observed behavior with the latest 5.3 toolchain on macOS:
```swift
$ swift
Welcome to Apple Swift version 5.3 (swiftlang-1200.0.28.1 clang-1200.0.30.1).
Type :help for assistance.
  1> import Foundation
  2> struct Foo: Codable {
  3.     private enum CodingKeys: CodingKey { case a }
  4.     public init(from decoder: Decoder) throws {
  5.         var container = try decoder.container(keyedBy: CodingKeys.self)
  6.         var aContainer = try container.nestedUnkeyedContainer(forKey: .a)
  7.         while true {
  8.             do { print(try aContainer.decode(Int.self)) }
  9.             catch { print(error); break }
 10.         }
 11.     }
 12.     public func encode(to encoder: Encoder) throws { fatalError() }
 13. }
 14> let f = try JSONDecoder().decode(Foo.self, from: "{\"a\":[1,2,3]}".data(using: .utf8)!)
1
2
3
valueNotFound(Swift.Int, Swift.DecodingError.Context(codingPath: [_JSONKey(stringValue: "Index 3", intValue: 3)], debugDescription: "Unkeyed container is at end.", underlyingError: nil))
f: Foo = {}
(Foo)  = {}
```
The behavior was identical when I tested with 5.2 toolchains on Linux.

I'll update the PR to throw the `valueNotFound` error instead 🙂